packages/ra-core/src/form/useSuggestions.ts

```diff
@@ -169,15 +174,15 @@ export const getSuggestionsFactory = ({
             // ignore the filter to show more choices
             suggestions = removeAlreadySelectedSuggestions(
                 choices,
-                selectedItem,
+                allowDuplicates ? null : selectedItem,
```

Comment:
if `allowDuplicates` is true, `removeAlreadySelectedSuggestions()` returns the input unchanged, so I'd prefer that you test it it's true on lone 175 and avoid calling `removeAlreadySelectedSuggestions` altogether in that case.